### PR TITLE
[flutter_tools] let experimental compiler support plugins

### LINKED ISSUE #44230: Find a solution for web plugins on experimental incremental compiler
Not using build_runner means we need to do some more work for plugin registration.

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Generate an entrypoint file when using the experimental compiler, containing the plugin import as well as the platform bootstrap.

Fixes https://github.com/flutter/flutter/issues/44230

### CODE REVIEW

**Review by yjbanov — APPROVED** [review]
LGTM if LGT @hterkelsen and @ditman

File: packages/flutter_tools/lib/src/web/bootstrap.dart [hunk 1]
```diff
@@ -55,7 +55,7 @@ window.\$hotReloadHook = function(modules) {
         // once we've reloaded every module, trigger the hot reload.
         if (reloadCount == modules.length) {
           require(["$entrypoint", "dart_sdk"], function(app, dart_sdk) {
-            window.\$mainEntrypoint = app.main.main;
+            window.\$mainEntrypoint = app[Object.keys(app)[0]].main;
```
Comment (hterkelsen) [inline packages/flutter_tools/lib/src/web/bootstrap.dart]: I'm not really familiar with RequireJS/AMD modules so it's not clear to my why the "app" dependency changed based on the changes in this PR... Is it the case that "app" will only ever have one property but we don't know the name of it at compile time? Please add a comment explaining why it's written this way

**Review by harryterkelsen — APPROVED** [review]
LGTM except for one change I think would be clearer with documentation

File: packages/flutter_tools/lib/src/web/bootstrap.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/web/bootstrap.dart]: Added a comment below. Essentially the module name is based off of the file, which for the generated entrypoint is something like `__var__flutter_123abcd__entrypoint`

**Pull request merged**